gateway/src/main.rs

```diff
@@ -32,6 +36,54 @@ async fn main() -> std::io::Result<()> {
         exit(1);
     });
 
+    let provider: Option<TracerProvider> = match config.clone().telemetry {
```

Comment:
can you move it into the file where we do config parsing? `main()` is cluttered enough already